pyproject.toml

```diff
@@ -0,0 +1,183 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+name = "g2p"
+dynamic = ["version"]
+description = "Module for creating context-aware, rule-based G2P mappings that preserve indices"
+readme = "README.md"
+license = "MIT"
+requires-python = ">= 3.7"
+authors = [
+    { name = "Aidan Pine", email = "hello@aidanpine.ca" },
+]
+classifiers = [
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+]
+dependencies = [
+    "click>=8.0.4",
+    "coloredlogs>=15.0.1",
+    "networkx>=2.6",
+    "openpyxl",
+    "panphon>=0.19",
+    "pydantic>=2.3",
+    "pyyaml>=5.2",
+    "regex",
+    "text_unidecode",
+    "tqdm",
+]
+
+[project.optional-dependencies]
+api = [
+    "eventlet>=0.33.0,!=0.36.0",
+    "flask-cors>=3.0.9",
+    "flask-restful>=0.3.9",
+    "flask-talisman>=0.7.0",
+    "flask<2.3; python_version < '3.8'",
+    "flask; python_version >= '3.8'",
+    "flask_socketio>=5.0.0",
+    "python-engineio>=4.0.0",
+    "python-socketio>=5.0.0",
+]
+test = [
+  "g2p[api]",
+  "coverage[toml]>=6.5",
+  "playwright>=1.26.1",
+  "jsonschema>=4.17.3",
+  "pep440>=0.1.2",
+]
+dev = [
+  "g2p[test]",
+  "black>=23; python_version < '3.8'",
+  "black>24.2; python_version >= '3.8'",
```

Comment:
Just noticed this small problem: black should be `^24.3`, not `>24.2`: we don't want to get black 25 by accident, since black promises not to change code formatting rules for a given major, but usually modifies the rules when they bump the major.
To avoid format flip-flopping issues, we want a PR to change the major and apply it to the whole code base all at once, not accidentally get it in some but not all sandboxes.

Reply:
Does Python support `^` as a version specifier?  I think it needs to be `>24.2,<25`.

Reply:
Yeah, actually it needs to be `~=24.2`: https://peps.python.org/pep-0440/#version-specifiers